### PR TITLE
Feat test splitting

### PR DESCRIPTION
**Pull request by dstallenberg**
* added test splitting as post processing step
* added meta comment adding as post processing step
* added test de-duplication as post processing step (using secondary objectives)

### CODE REVIEW

File: libraries/search-javascript/lib/testbuilding/JavaScriptDecoder.ts [hunk 1]
```diff
@@ -128,10 +127,10 @@ export class JavaScriptDecoder implements Decoder<JavaScriptTestCase, string> {
       "// Imports",
       ...imports,
       gatherAssertionData ? assertionFunction : "",
-      `describe('${targetName}', function() {`,
+      `describe('TODO proper test naming', function() {`,
```
Comment (mitchellolsthoorn) [inline libraries/search-javascript/lib/testbuilding/JavaScriptDecoder.ts:130]: Is this name overwritten somewhere else? This should not be in here

File: tools/javascript/lib/workflows/MetaComment.ts [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+//               }
+//               encoding.addMetaComment("");
+//             } else {
+//               encoding.addMetaComment(
+//                 `Selected for objective: ${objective.getIdentifier()}`
+//               );
+//             }
+//           }
+//     }
+// }
```
Comment (mitchellolsthoorn) [inline tools/javascript/lib/workflows/MetaComment.ts:101]: Is this still needed?

**Review by mitchellolsthoorn — CHANGES_REQUESTED** [review]

File: libraries/search-javascript/lib/testbuilding/JavaScriptDecoder.ts (hunk 1 above)
Comment (dstallenberg) [reply libraries/search-javascript/lib/testbuilding/JavaScriptDecoder.ts:130]: This was necessary for refactoring reasons.
In a later PR the test name should be in the encoding.
Also the "targetName" was not really descriptive anyway.

File: tools/javascript/lib/workflows/MetaComment.ts (hunk 2 above)
Comment (dstallenberg) [reply tools/javascript/lib/workflows/MetaComment.ts:101]: Nope. Removed it.

File: libraries/search-javascript/lib/testbuilding/JavaScriptDecoder.ts (hunk 1 above)
Comment (mitchellolsthoorn) [reply libraries/search-javascript/lib/testbuilding/JavaScriptDecoder.ts:130]: Isn't is better to put something other than this in for now at least?

Comment (dstallenberg) [reply libraries/search-javascript/lib/testbuilding/JavaScriptDecoder.ts:130]: Any suggestions?

Comment (dstallenberg) [reply libraries/search-javascript/lib/testbuilding/JavaScriptDecoder.ts]: Changed to SynTest Test Suite.

**Review by mitchellolsthoorn — APPROVED** [review]

**Pull request merged**